src/common/utxobased/network/blockbookApi.ts

```diff
@@ -85,22 +95,36 @@ export const asBlockbookTransaction = (
     blockTime: asNumber,
     fees: asString,
     vin: asArray(
-      asObject({
-        txid: asString,
-        // Empirically observed omitted sequence is possible for when sequence is zero.
-        // Is the case for tx `19ecc679cfc7e71ad616a22bbee96fd5abe8616e4f408f1f5daaf137400ae091`.
-        sequence: asOptional(asNumber, 0),
-        n: asNumber,
-        // If Blockbook doesn't provide vout, assume 0. Empirically observed
-        // case for tx `fefac8c22ba1178df5d7c90b78cc1c203d1a9f5f5506f7b8f6f469fa821c2674`
-        // which has no `vout` for input in WebSocket response payload but block
-        // will show the input's vout value to be `0`.
-        vout: asOptional(asNumber, 0),
-        addresses: asArray(asAddress),
-        isAddress: asBoolean,
-        value: asString,
-        hex: asOptional(asString)
-      })
+      asEither(
+        // Address input:
+        asObject({
```

Comment:
Optional: This cleaner is getting pretty deep. It might be easier to read if we define `asAddressVin` and `asCoinbaseVin` types on the outside, and then do `asEither(asAddressVin, asCoinbaseVin)` in here.